File: migrate_data.py
#!/usr/bin/env python3

import pandas as pd
import firebase_admin
from firebase_admin import credentials, db
import os

# TODO: Replace with the path to your service account key file
cred = credentials.Certificate("Service Accounts/inventory-and-sales-59f32-firebase-adminsdk-fbsvc-d2d01545d8.json")

# TODO: Replace with your Firebase project's database URL
firebase_admin.initialize_app(cred, {
    'databaseURL': "https://inventory-and-sales-59f32-default-rtdb.firebaseio.com"
})

# List of CSV files to migrate
csv_files = [
    'Service Accounts/CSV Files/Inventory Management - Inventory Count Log.csv',
    'Service Accounts/CSV Files/Inventory Management - Sales.csv'
]

for csv_file_path in csv_files:
    # Get the base name of the file to use as the db node
    db_node = os.path.splitext(os.path.basename(csv_file_path))[0]

    # Read data from CSV file
    df = pd.read_csv(csv_file_path)

    # Convert dataframe to a list of dictionaries
    data = df.to_dict(orient='records')

    # Get a reference to the database node
    ref = db.reference(db_node)

    # Push data to the database
    for item in data:
        ref.push().set(item)

    print(f"Data from {csv_file_path} migrated successfully to node '{db_node}'!")

File: migrate-data.js

import admin from 'firebase-admin';
import fs from 'fs';
import csv from 'csv-parser';
import { createRequire } from 'module';
const require = createRequire(import.meta.url);

// TODO: Replace with the path to your service account key file
const serviceAccount = require('./Service Accounts/inventory-and-sales-59f32-firebase-adminsdk-fbsvc-d2d01545d8.json');

// TODO: Replace with your Firebase project's database URL
admin.initializeApp({
  credential: admin.credential.cert(serviceAccount),
  databaseURL: 'https://inventory-and-sales-59f32-default-rtdb.firebaseio.com'
});

const db = admin.database();

// List of CSV files to migrate
const csvFiles = [
  'Service Accounts/CSV Files/Inventory Management - Inventory Count Log.csv',
  'Service Accounts/CSV Files/Inventory Management - Kampstore Sales.csv'
];

csvFiles.forEach(csvFilePath => {
  const dbNode = csvFilePath.split('/').pop().split('.')[0];
  const ref = db.ref(dbNode);

  fs.createReadStream(csvFilePath)
    .pipe(csv())
    .on('data', (row) => {
      const sanitizedRow = {};
      for (const key in row) {
        const sanitizedKey = key.replace(/\//g, '_');
        sanitizedRow[sanitizedKey] = row[key];
      }
      ref.push(sanitizedRow);
    })
    .on('end', () => {
      console.log(`Data from ${csvFilePath} migrated successfully to node '${dbNode}'!`);
    });
});
